web/src/components/storage/utils.test.js

```diff
@@ -44,3 +44,44 @@ describe("deviceLabel", () => {
     expect(result).toEqual("/dev/sda");
   });
 });
+
+describe("parseSize", () => {
+  it("returns bytes from given input", () => {
+    expect(parseSize(1024)).toEqual(1024);
+    expect(parseSize("1024")).toEqual(1024);
+    expect(parseSize("1 KiB")).toEqual(1024);
+    expect(parseSize("2 MiB")).toEqual(2097152);
+  });
+
+  it("returns 0 if given input is null, undefined, or empty string", () => {
+    expect(parseSize(null)).toEqual(0);
+    expect(parseSize(undefined)).toEqual(0);
+    expect(parseSize("")).toEqual(0);
+  });
+
+  it("does not include decimal part of resulting conversion", () => {
+    expect(parseSize("1024.32 KiB")).toEqual(1048903); // Not 1048903.68
+  });
+});
+
+describe("splitSize", () => {
+  it("returns a size object with size and unit from given input", () => {
+    expect(splitSize("2048 KiB")).toEqual({ size: 2048, unit: "KiB" });
+  });
+
+  it("returns a size object with the result of converting the input when no string is given", () => {
+    expect(splitSize(1000)).toEqual({ size: 1000, unit: "B" });
+    expect(splitSize(1024)).toEqual({ size: 1, unit: "KiB" });
+    expect(splitSize(1048576)).toEqual({ size: 1, unit: "MiB" });
+    expect(splitSize(undefined)).toEqual({ size: 0, unit: "B" });
+    expect(splitSize(null)).toEqual({ size: 0, unit: "B" });
+  });
+
+  it("returns an 'empty' size object when -1 is given", () => {
+    expect(splitSize(-1)).toEqual({ size: "", unit: "" });
+  });
+
+  it("returns an 'empty' size object when empty string is given", () => {
+    expect(splitSize("")).toEqual({ size: "", unit: "" });
+  });
```

Comment:
What about `splitSize("30")`?

Reply:
I have _improved_ the `splitSize` code. Admittedly, it's a bit ugly function. I'm not happy with it, but I hope we can improve the situation in next iterations. Sorry, I have no better idea by now for dealing with sizes in this form.